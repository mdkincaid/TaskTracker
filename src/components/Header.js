// react
import PropTypes from 'prop-types'
import React from 'react'
import { useLocation } from 'react-router-dom'
import { FaMinus, FaPlus } from 'react-icons/fa'

const Header = ({ title, onAdd, showAdd }) => {
    const location = useLocation()
    let addButton

    if (location.pathname === '/' && !showAdd) {
        addButton = (
            <button aria-label="show add task" type="button">
                <FaPlus size={28} onClick={onAdd} cursor="pointer" />
            </button>
        )
    } else if (location.pathname === '/' && showAdd) {
        addButton = (
            <button aria-label="hide add task" type="button">
                <FaMinus size={28} onClick={onAdd} cursor="pointer" />
            </button>
        )
    }

    return (
        <header className="header">
            <h1>{title}</h1>
            {addButton}
        </header>
    )
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    onAdd: PropTypes.func.isRequired,
    showAdd: PropTypes.bool.isRequired,
}

export default Header
