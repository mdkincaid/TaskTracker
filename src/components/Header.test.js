// react
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// project
import Header from './Header';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useLocation: () => ({
		pathname: 'localhost:3000/example/path',
	}),
}));

describe('Header', () => {
	it('should render', () => {
		const result = render(<Header />);

		// assert
		expect(result.container).toMatchSnapshot();
	});

	describe('props', () => {
		it('should render a title of Task Tracker', () => {
			// arrange / act
			const title = 'Task Tracker';
			const result = render(<Header>{title}</Header>);

			// assert
			expect(result.getByText(title)).toBeTruthy()
		});

		// it('should render the FaPlus icon if showAdd is false', () => {
		// 	// arrange / act
		// 	const showAdd = false;
		// 	let wrapped = shallow(<Header showAdd={showAdd}></Header>);

		// 	// assert
		// 	expect(wrapped.find('FaPlus')).toBeDefined();
		// });

		// it('should render the FaMinus icon if showAdd is true', () => {
		// 	// arrange / act
		// 	const showAdd = true;
		// 	let wrapped = shallow(<Header showAdd={showAdd}></Header>);

		// 	// assert
		// 	expect(wrapped.find('FaMinus')).toBeDefined();
		// });
	});
});
