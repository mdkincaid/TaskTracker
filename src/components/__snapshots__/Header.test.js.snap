// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Test Title
    </h1>
    <button
      aria-label="show add task"
      type="button"
    >
      <svg
        cursor="pointer"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </header>
</div>
`;

exports[`Header should render the FaMinus icon if showAdd is true 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Test
    </h1>
    <button
      aria-label="hide add task"
      type="button"
    >
      <svg
        cursor="pointer"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </header>
</div>
`;

exports[`Header should render the FaPlus icon if showAdd is false 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Test
    </h1>
    <button
      aria-label="show add task"
      type="button"
    >
      <svg
        cursor="pointer"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </header>
</div>
`;
